src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,22 @@
+import Popup from "./Popup";
+
+class PopupWithConfirm extends Popup {
+  setSubmitAction(action) {
+    this._handleSubmitCallback = action;
+  }
+
+  open(id) {
+    super.open();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.